src/objects/impls/deposit.rs

```diff
@@ -0,0 +1,14 @@
+use crate::{constants::ResourceType, macros::*, objects::Deposit};
+
+simple_accessors! {
+    impl Deposit {
+        pub fn last_cooldown() -> u32 = lastCooldown;
+        pub fn ticks_to_decay() -> u32 = ticksToDecay;
```

Comment:
Usually handled by `CanDecay` trait.

Reply:
Good call - I duplicated that implementation from `Tombstone` so maybe it can be removed from there as well?